Sources/FoundationEssentials/String/String+Path.swift

```diff
@@ -369,33 +369,27 @@ extension String {
     #if !NO_FILESYSTEM
     internal static func homeDirectoryPath(forUser user: String? = nil) -> String {
 #if os(Windows)
-        func GetUserProfile() -> String? {
-            return "USERPROFILE".withCString(encodedAs: UTF16.self) { pwszVariable in
-                let dwLength: DWORD = GetEnvironmentVariableW(pwszVariable, nil, 0)
-                // Ensure that `USERPROFILE` is defined.
-                if dwLength == 0 { return nil }
-                return withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(dwLength)) {
-                    guard GetEnvironmentVariableW(pwszVariable, $0.baseAddress, dwLength) == dwLength - 1 else {
-                        return nil
-                    }
-                    return String(decodingCString: $0.baseAddress!, as: UTF16.self)
+        if let user {
+            func fallbackUserDirectory() -> String {
+                guard let systemDrive = ProcessInfo.processInfo.environment["SystemDrive"] else {
```

Comment:
I think we should use `ALLUSERSPROFILE` instead (https://learn.microsoft.com/en-us/windows/deployment/usmt/usmt-recognized-environment-variables).